### DISCUSSION TITLE
Installing Homebrew sees wrong Git

### DISCUSSION THREAD

**Discussion by efweber999** [Homebrew/discussions#1547] [Getting started] [answered]
Hi,

I'm trying to install Homebrew on an AWS EC2 instance running Centos7. The installation failed saying that my versions of curl and git were too old. I've updated both the the most recent stable versions. I deleted all files from the failed installation attempt. I shut down and restarted the EC2 instance. When I check the git installation I get the the following which looks fine.

```
$ echo $PATH
/opt/rh/devtoolset-8/root/usr/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/home/centos/.local/bin:/home/centos/bin
$ which git
/usr/local/bin/git
$ git --version
git version 2.31.1
```

When I try and install homebrew:

`/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"`

it still gives me:

```
HEAD is now at 46f817807ae gcc: update 11.1.0_1 bottle.
Error: Please update your system Git.
Minimum required version: 2.7.0
Your Git version: 1.8.3.1
Your Git executable: /usr/bin/git
```

Why isn't it respecting the PATH variable? What am I missing?

Thanks

**Comment by SMillerDev** [comment]
Pinging @Homebrew/linux for help

**Comment by carlocab** [comment, marked as answer]
This comes from `brew`'s environment filtering: it ignores most of your `PATH` (and most other environment variables) on purpose.

If you want it to use your newer `git`, you can do `export HOMEBREW_NO_ENV_FILTERING=1`, but this will show you a deprecation warning. An alternative is to `export HOMEBREW_DEVELOPER=1` and set `HOMEBREW_GIT` to the path to your `git` executable.

**Reply by efweber999** [reply]
Thanks!

**Comment by gromgit** [comment]
You didn't show the full output of the install, but correlating the messages you _did_ include with the actual source of `install.sh` suggests that your core tap has been pulled down correctly, so try this one-liner:
```
HOMEBREW_NO_ENV_FILTERING=1 brew install git
```
This installs a Homebrew Git bottle, which _should_ thereafter be automagically used since, as the `brew` man page says:

> **ENVIRONMENT**
> ...
>        `HOMEBREW_FORCE_BREWED_GIT`
>               If set, always use a Homebrew-installed `git`(1) rather than the system version. **Automatically set if the system version of `git` is too old.**

**Reply by efweber999** [reply]
Thanks!
